ambry-cloud/src/main/java/com/github/ambry/cloud/HelixVcrClusterParticipant.java

```diff
@@ -105,21 +117,44 @@ public HelixVcrClusterParticipant(CloudConfig cloudConfig, ClusterMapConfig clus
    * @param partitionIdStr The partitionIdStr notified by Helix.
    */
   public void addPartition(String partitionIdStr) {
-    PartitionId partitionId = partitionIdMap.get(partitionIdStr);
+    // Add partition with retry logic in case Ambry clustermap is not update to date.
+    retryExecutor.runWithRetries(retryPolicy, callback -> doAddPartition(partitionIdStr, callback), e -> true,
+        (result, exception) -> {
+          if (exception != null) {
+            logger.warn("Partition {} is not in clusterMap on addPartition and retry failed.", partitionIdStr,
```

Comment:
I guess we should also log a error message and a metric when we exhaust all the retries. Because we have no other way of knowing if a partition update totally failed.

Reply:
Also are we assuming that the only exception possible is partition not in clustemap?

Reply:
The actual vcr cluster update is a scheduled task(`updateVcrHelix`), so we can't know its result here.  The logic here is only to trigger the update.

Added an exception catch in `updateVcrHelix`.

All exception should be retried. Updated the log here.